include/RAJA/pattern/params/reducer.hpp

```diff
@@ -83,16 +55,41 @@ namespace detail
   // Basic Reducer
   //
   //
-  template <typename Op, typename T>
+
+  // Basic data type Reducer
+  // T must be a basic data type
+  // VType must be ValOp<T, Op>
+  template <typename Op, typename T, typename VType>
   struct Reducer : public ForallParamBase {
     using op = Op;
-    using value_type = T;
+    using value_type = T; // This is a basic data type
 
-    RAJA_HOST_DEVICE Reducer() {}
-    Reducer(value_type *target_in) : target(target_in), val(op::identity()) {}
+    Reducer() = default;
 
+    // Basic data type constructor
+    RAJA_HOST_DEVICE Reducer(value_type *target_in) : m_valop(VType{}), target(target_in){}
```

Comment:
I don't think you have to explicitly create a VType here, its already done int he declaration of m_valop.

Reply:
Thanks for pointing this out. It won't hurt to leave this in, and it makes it clear what we're doing, doubly so.